hazelcast/src/main/java/com/hazelcast/jet/impl/connector/StreamEventJournalP.java

```diff
@@ -344,39 +352,40 @@ private CompletableFuture<? extends ReadResultSet<? extends T>> readFromJournal(
                                  .toCompletableFuture();
     }
 
-    private static class ClusterMetaSupplier<E, T> implements ProcessorMetaSupplier {
+    public static HzClientDataLinkFactory getDataStoreFactory(HazelcastInstance hazelcastInstance, String name) {
+        NodeEngineImpl nodeEngine = Util.getNodeEngine(hazelcastInstance);
+        DataLinkService dataLinkService = nodeEngine.getDataLinkService();
+        DataLinkFactory<?> dataLinkFactory = dataLinkService.getDataLinkFactory(name);
+
+        if (!(dataLinkFactory instanceof HzClientDataLinkFactory)) {
+            String className = HzClientDataLinkFactory.class.getSimpleName();
+            throw new HazelcastException("Data store factory '" + name + "' must be an instance of " + className);
+        }
+        return (HzClientDataLinkFactory) dataLinkFactory;
+    }
+
+    // Creates ClusterProcessorSupplier per member. Each ClusterProcessorSupplier is given a
+    // list of partitions IDs
+    private static class ClusterMetaSupplier<E> implements ProcessorMetaSupplier {
```

Comment:
This lost the type parameter `T`, Constructor takes `ClusterMetaSupplierParams` but without types.

Also, there were some renames of the type parameters, but now they don't match with the public API, it makes it confusing. It's best to avoid such unnecessary changes.

Reply:
Added the parameter T. Also renamed all type parameters according to public API.